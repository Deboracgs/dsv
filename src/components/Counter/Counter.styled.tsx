import styled from "styled-components";

export const CounterContainerStyled = styled.div`
    display: inline-flex;
    buttonÂ {
        height: 56px;
        max-height: 100%;
    }
`

export const InputStyled = styled.div`
    width: 50%;
    display: inline-flex;
`

