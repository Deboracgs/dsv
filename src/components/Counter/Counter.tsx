import { Box } from "@mui/system";
import { Grid, Typography } from "@mui/material";
import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { commonSelector, updateCounter } from "./../../effects/reducers";
import { CounterContainerStyled, InputStyled,  } from "./Counter.styled";
import { Button, InputNumber } from "./../index"
import { useEffect } from "react";


export interface CounterProps {
}

const modes = ["normal", "odd", "input", "reset"]
const Counter = (): React.ReactElement => {
    const dispatch = useDispatch();
    const [mode, setMode] = useState("normal");
    const [valueToAdd, setValueToAdd] = useState(0);
    const { counter } = useSelector(commonSelector);
   
    useEffect(() => { 
        if(mode === "reset"){
            setMode("normal");
            setValueToAdd(0)
            dispatch(updateCounter(0));
        }
    }, [mode, dispatch]);

    const nextOdd = () => {
        let newValue = counter;
        if(counter % 2 !== 0) {
            newValue = counter + 2;
        }else {
            newValue = counter + 1;
        }

        dispatch(updateCounter(newValue));
    }

    const previousOdd = () => {
        let newValue = counter;
        if(counter % 2 !== 0) {
            newValue = counter - 2;
        }else {
            newValue = counter - 1;
        }

        dispatch(updateCounter(newValue <= 0 ? 1 : newValue));
    }

    const inputMode = (type: string) => {
        let newValue = counter;
        if(type === "increment"){
            newValue = counter + Number(valueToAdd);
        }else{
            newValue = counter - Number(valueToAdd);
        }
        dispatch(updateCounter(newValue <= 0 ? 0 : newValue));
    }

    const decrementCounter = () => {
        switch (mode) {
            case "odd":
                previousOdd()
                break;

            case "input":
                inputMode("decrement")
                break;
        
            default:
                dispatch(updateCounter(counter === 0 ? 0 : counter -1));
                break;
        }
       
    }

    const incrementCounter =  () => {
        switch (mode) {
            case "odd":
                nextOdd()
                break;

            case "input":
                inputMode("increment")
                break;
        
            default:
                dispatch(updateCounter(counter + 1));
                break;
        }
       
    }

    return (
        <Box>
            <Grid>
                <Grid item>
                    <Typography variant="h3">Counter: <span data-testid="counter-value">{counter}</span></Typography>
                </Grid>
                <Grid item>
                    <CounterContainerStyled>
                        <Button
                            id="decrement"
                            label="-"
                            onClick={() => decrementCounter()}
                        />
                        <InputStyled>
                            <InputNumber
                                id="counter"
                                defaultValue={valueToAdd}
                                disabled={!(mode === "input")}
                                onChange={(value: number) => setValueToAdd(value)}
                            />
                        </InputStyled>
                        
                        <Button
                            id="increment"
                            onClick={() => incrementCounter()}
                            label="+"
                        />
                    </CounterContainerStyled>
                </Grid>
                <Grid item>
                    <Grid container spacing={2}>
                        {modes.map((mode) => (
                            <Grid item marginY={2} key={mode}>
                                <Button
                                    id={mode}
                                    variant="outlined"
                                    label={`${mode} Mode`}
                                    onClick={() => setMode(mode)}
                                />
                            </Grid>
                        ))}
                    
                    </Grid>
                </Grid>
            </Grid>
        </Box>
       
    )
};

Counter.defaultProps = {};
export default Counter;
