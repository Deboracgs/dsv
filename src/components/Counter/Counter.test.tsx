import { fireEvent, render } from "@testing-library/react";
import WrapperMock from "../../utils/wrapperMock";
import Counter, { CounterProps } from "./Counter";

describe("Counter Component", () => {
  const defaultProps: CounterProps = {
  };

  const counterId = "counter-value";

  const rendered = (props: CounterProps) =>
    render(
      <WrapperMock>
        <Counter {...props} />
      </WrapperMock>
    );

  it("render", () => {
    const { getByTestId, asFragment } = rendered(defaultProps);

    const element = getByTestId(counterId);

    expect(element).toBeDefined();
    expect(element.textContent).toBe("0");
    expect(asFragment()).toMatchSnapshot();
  });

  it("Normal Mode", () => {
    const { getByTestId, asFragment } = rendered(defaultProps);

    const counter = getByTestId(counterId);
    const buttonIncrement = getByTestId("button-increment");
    const buttonDecrement = getByTestId("button-decrement");
    const element = getByTestId("button-normal");
    fireEvent.click(element);
    fireEvent.click(buttonIncrement);
    expect(counter.textContent).toBe("1");
    fireEvent.click(buttonDecrement);
    expect(counter.textContent).toBe("0");

    expect(asFragment()).toMatchSnapshot();
  });

  it("ODD Mode", () => {
    const { getByTestId, asFragment } = rendered(defaultProps);

    const counter = getByTestId(counterId);
    const buttonIncrement = getByTestId("button-increment");
    const buttonDecrement = getByTestId("button-decrement");
    const element = getByTestId("button-odd");
    fireEvent.click(element);
    fireEvent.click(buttonIncrement);
    expect(counter.textContent).toBe("1");
    fireEvent.click(buttonIncrement);
    expect(counter.textContent).toBe("3");
    fireEvent.click(buttonDecrement);
    expect(counter.textContent).toBe("1");

    expect(asFragment()).toMatchSnapshot();
  });

  it("Input Mode", () => {
    const { getByTestId, asFragment, container } = rendered(defaultProps);

    const counter = getByTestId(counterId);
    const fieldText = container.querySelector("#counter") as HTMLInputElement;
    const buttonIncrement = getByTestId("button-increment");
    const buttonDecrement = getByTestId("button-decrement");
    const element = getByTestId("button-input");
    fireEvent.click(element);
    fireEvent.change(fieldText, {target: {value: "3"}});

    fireEvent.click(buttonIncrement);
    expect(counter.textContent).toBe("4");
    fireEvent.click(buttonIncrement);
    expect(counter.textContent).toBe("7");
    fireEvent.click(buttonDecrement);
    expect(counter.textContent).toBe("4");
    fireEvent.click(buttonDecrement);
    expect(counter.textContent).toBe("1");

    expect(asFragment()).toMatchSnapshot();
  });

  it("Reset Mode", () => {
    const { getByTestId, asFragment } = rendered(defaultProps);

    const counter = getByTestId(counterId);
    const buttonIncrement = getByTestId("button-increment");
    const element = getByTestId("button-reset");

    fireEvent.click(buttonIncrement);
    expect(counter.textContent).toBe("2");

    fireEvent.click(element);

    expect(counter.textContent).toBe("0");   
    expect(asFragment()).toMatchSnapshot();
  });

});
