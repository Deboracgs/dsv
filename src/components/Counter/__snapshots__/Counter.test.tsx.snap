// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Input Mode 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Counter: 
          <span
            data-testid="counter-value"
          >
            1
          </span>
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="sc-bczRLJ IfZwI"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-decrement"
            id="button-decrement"
            tabindex="0"
            type="button"
          >
            -
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="sc-gsnTZi bvslhN"
          >
            <div
              data-testid="input-number-counter"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="counter"
                    name="input-number-counter"
                    type="number"
                    value="3"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-increment"
            id="button-increment"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-normal"
              id="button-normal"
              tabindex="0"
              type="button"
            >
              normal Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-odd"
              id="button-odd"
              tabindex="0"
              type="button"
            >
              odd Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-input"
              id="button-input"
              tabindex="0"
              type="button"
            >
              input Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-reset"
              id="button-reset"
              tabindex="0"
              type="button"
            >
              reset Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Counter Component Normal Mode 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Counter: 
          <span
            data-testid="counter-value"
          >
            0
          </span>
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="sc-bczRLJ IfZwI"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-decrement"
            id="button-decrement"
            tabindex="0"
            type="button"
          >
            -
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="sc-gsnTZi bvslhN"
          >
            <div
              data-testid="input-number-counter"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled=""
                    id="counter"
                    name="input-number-counter"
                    type="number"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-increment"
            id="button-increment"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-normal"
              id="button-normal"
              tabindex="0"
              type="button"
            >
              normal Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-odd"
              id="button-odd"
              tabindex="0"
              type="button"
            >
              odd Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-input"
              id="button-input"
              tabindex="0"
              type="button"
            >
              input Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-reset"
              id="button-reset"
              tabindex="0"
              type="button"
            >
              reset Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Counter Component ODD Mode 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Counter: 
          <span
            data-testid="counter-value"
          >
            1
          </span>
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="sc-bczRLJ IfZwI"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-decrement"
            id="button-decrement"
            tabindex="0"
            type="button"
          >
            -
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="sc-gsnTZi bvslhN"
          >
            <div
              data-testid="input-number-counter"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled=""
                    id="counter"
                    name="input-number-counter"
                    type="number"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-increment"
            id="button-increment"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-normal"
              id="button-normal"
              tabindex="0"
              type="button"
            >
              normal Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-odd"
              id="button-odd"
              tabindex="0"
              type="button"
            >
              odd Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-input"
              id="button-input"
              tabindex="0"
              type="button"
            >
              input Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-reset"
              id="button-reset"
              tabindex="0"
              type="button"
            >
              reset Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Counter Component Reset Mode 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Counter: 
          <span
            data-testid="counter-value"
          >
            0
          </span>
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="sc-bczRLJ IfZwI"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-decrement"
            id="button-decrement"
            tabindex="0"
            type="button"
          >
            -
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="sc-gsnTZi bvslhN"
          >
            <div
              data-testid="input-number-counter"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled=""
                    id="counter"
                    name="input-number-counter"
                    type="number"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-increment"
            id="button-increment"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-normal"
              id="button-normal"
              tabindex="0"
              type="button"
            >
              normal Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-odd"
              id="button-odd"
              tabindex="0"
              type="button"
            >
              odd Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-input"
              id="button-input"
              tabindex="0"
              type="button"
            >
              input Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-reset"
              id="button-reset"
              tabindex="0"
              type="button"
            >
              reset Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Counter Component render 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Counter: 
          <span
            data-testid="counter-value"
          >
            0
          </span>
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="sc-bczRLJ IfZwI"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-decrement"
            id="button-decrement"
            tabindex="0"
            type="button"
          >
            -
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="sc-gsnTZi bvslhN"
          >
            <div
              data-testid="input-number-counter"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled=""
                    id="counter"
                    name="input-number-counter"
                    type="number"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-increment"
            id="button-increment"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-normal"
              id="button-normal"
              tabindex="0"
              type="button"
            >
              normal Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-odd"
              id="button-odd"
              tabindex="0"
              type="button"
            >
              odd Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-input"
              id="button-input"
              tabindex="0"
              type="button"
            >
              input Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-86otrc-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              data-testid="button-reset"
              id="button-reset"
              tabindex="0"
              type="button"
            >
              reset Mode
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
